control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs

```diff
@@ -126,6 +132,201 @@ async fn nexus_reconciler(
     }
 }
 
+#[tracing::instrument(skip(nexus, context), level = "trace", fields(nexus.uuid = %nexus.uuid(), request.reconcile = true))]
+pub(super) async fn handle_faulted_child(
+    nexus: &mut OperationGuardArc<NexusSpec>,
+    context: &PollContext,
+) -> PollResult {
+    info!("entered handle faulted child");
+    let nexus_uuid = nexus.uuid();
+    let nexus_state = context.registry().nexus(nexus_uuid).await?;
+    let child_count = nexus_state.children.len();
+    if nexus_state.status == NexusStatus::Degraded && child_count > 1 {
+        let nexus_spec_clone = nexus.lock().clone();
+        for child in nexus_state.children.iter().filter(|c| c.state.faulted()) {
+            if !nexus_spec_clone.rebuild_state.contains_key(&child.uri) {
+                let wait_time = RuleSet::faulted_child_wait(&nexus_state);
+                if wait_time.is_zero() {
+                    info!("Start full rebuild");
+                    nexus
+                        .set_rebuild_state(
+                            context.registry(),
+                            child.uri.clone(),
+                            Some(RebuildInfo::new(None, RebuildStage::FullRebuildInit)),
+                        )
+                        .await?;
+                } else {
+                    info!("Setting rebuild state as Timed Wait for child");
+                    nexus
+                        .set_rebuild_state(
+                            context.registry(),
+                            child.uri.clone(),
+                            Some(RebuildInfo::new(Some(wait_time), RebuildStage::TimedWait)),
+                        )
+                        .await?;
+                }
+            }
+        }
+    }
+    PollResult::Ok(PollerState::Idle)
+}
+
+async fn get_child_node(
+    nexus: NexusSpec,
+    child: &Child,
+    context: &PollContext,
+) -> Result<NodeId, SvcError> {
+    for nexus_child in nexus.children.iter() {
+        if let Some(replica) = nexus_child.as_replica() {
+            if replica.uri() == &child.uri {
+                let replica = context.registry().get_replica(replica.uuid()).await?;
+                return Ok(replica.node);
+            } else {
+                continue;
+            }
+        }
+    }
+    Err(SvcError::NotFound {
+        kind: ResourceKind::Node,
+        id: "Child node not found".to_string(),
+    })
+}
+
+pub(super) async fn initialize_partial_rebuild(
+    nexus: &mut OperationGuardArc<NexusSpec>,
+    context: &PollContext,
+) -> PollResult {
+    info!("entered initialize partial rebuild function");
+    let nexus_uuid = nexus.uuid().clone();
+    let nexus_state = context.registry().nexus(&nexus_uuid).await?;
+    let nexus_spec = context.specs().nexus(&nexus_uuid).await?;
+    let node = context.registry().node_wrapper(&nexus_state.node).await?;
+    let rebuild_state = nexus_spec.lock().rebuild_state.clone();
+    let nexus_name = nexus_spec.lock().clone().name;
```

Comment:
Since we using nexus _name just for tracing, can remove this and use nexus_uuid itself to avoid a lock call.

Reply:
Will do